src/settingsview.cpp

```diff
@@ -156,7 +163,31 @@ void SettingsView::setReopenTab(bool reopen)
 
 void SettingsView::onDownloadDirChanged(const QString &dir)
 {
-    ui->downloadDirPath->setText(dir);
+    ui->downloadDirPath->setText(formatDownloadDir(dir));
+    ui->downloadDirPath->setToolTip(dir);
+}
+
+void SettingsView::onDownloadPathClicked()
+{
+    QString downloadPath = KiwixApp::instance()->getSettingsManager()->getDownloadDir();
+    QApplication::clipboard()->setText(downloadPath);
+
+    dissapearingPopup("Path was copied.");
+}
+
+void SettingsView::dissapearingPopup(QString message) 
+{
+    QMessageBox *popup = new QMessageBox();
+
+    popup->setAttribute(Qt::WA_DeleteOnClose);
+    popup->setText(message);
+    popup->setStandardButtons(QMessageBox::NoButton);
+
+    popup->move(QGuiApplication::primaryScreen()->geometry().center());
+
+    popup->show();
+
+    QTimer::singleShot(1250, popup, &QObject::deleteLater);
 }
```

Comment:
A popup window doesn't look nice. Couldn't it be implemented as a tooltip (near the click position rather than in the center of the screen)?

Reply:
I am not sure of what you are describing. Do you have a specific class/concept in mind?

Reply:
https://doc.qt.io/qt-5/qtooltip.html

Reply:
Tried something like this before but could not get it to display for some reason:
```
QPoint globalPos = ui->downloadDirPathCopy->mapToGlobal(QPoint(0, 0));
QToolTip::showText(globalPos, "test", ui->downloadDirPathCopy); 
```

Reply:
There are two issues that hinder the display of the tooltip:

1. [Make sure the tooltip is diplayed on mouse release event (rather than press event)](https://forum.qt.io/topic/139220/qtooltip-showtext-doesn-t-work-in-graphicsview-click-handler/4?_=1718886680667&lang=en-US)
2. The tooltip isn't diplayed if it would contain the mouse pointer position - you can test this by clicking on different points of the copy button/label. You won't have this problem if you call `QToolTip::showText()` with the current position of the mouse.

Reply:
Wow, this makes perfect sense. Thank you.